### PR TITLE
Modify appengine-web.xml file to deploy maven project

### PR DESCRIPTION
**Pull request by essencec**

### CODE REVIEW

**Review by ajwang0 — DISMISSED** [review]

File: client/dispatch.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# Rules for dispatch file...
+# Put specific cases at the top.
+dispatch:
+# Route the urls that point to the Java backend's API calls
+ - url: "*/api/v1/*"
+   service: backend
+# Route all other urls to the React.js frontend
+ - url: "*/*"
+   service: default
+
```
Comment (tonymai) [inline client/dispatch.yaml:11]: add a new line here

File: client/src/App.js [hunk 2]
```diff
@@ -1,24 +1,25 @@
...
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
+class App extends Component {
+    componentDidMount() {
+        console.log("Inside componentDidMount!");
```
Comment (tonymai) [inline client/src/App.js]: remove console log

File: client/src/App.js [hunk 3]
```diff
@@ -1,24 +1,25 @@
...
+        fetch("/api/v1/test-servlet")
+          .then((response) => response.text())
+          .then((text) => {
+        console.log("here is the text from servlet: ", text);
```
Comment (tonymai) [inline client/src/App.js]: the indentation looks off here

**Review by tonymai — DISMISSED** [review]
Overall nice!

**Review by ajwang0 — CHANGES_REQUESTED** [review]
https://github.com/googleinterns/play-web-step-2020/blob/master/client/src/App.test.js 

Can you also update the test

**Comment by essencec** [comment]
> https://github.com/googleinterns/play-web-step-2020/blob/master/client/src/App.test.js
> 
> Can you also update the test

Do you want me to create a test for promise?

File: client/src/App.js [hunk 4]
```diff
@@ -1,24 +1,24 @@
...
+                console.log("here is the text from servlet: ", text);
+            });
+    }
+
+    render() {
+        return (
+            <div className="App">
+                <div className="App-header">
+                    <h2>Welcome to React</h2>
+                </div>
```
Comment (ajwang0) [inline client/src/App.js:26]: Maybe instead we can add the text from the server onto this view to verify its working.

```
constructor(props) {
  super(props);
  this.state = { text: "" };
}

componentDidMount() {
        fetch("/api/v1/test-servlet")
            .then((response) => response.text())
            .then((text) => {
                this.setState({text});
            });
    }

...

render() {
   return (<div className="App">
                <div className="App-header">
                    <h2>Welcome to React</h2>
                   {this.state.text}
                </div>
      </div>
   );
}
```

This way we can both verify the fetch is actually hitting the backend correctly and it'll be a good reference for others when we want to replace it with the actual clusters api.

We can also test the componentDidMount is working by verifying the text appears

**Review by ajwang0 — CHANGES_REQUESTED** [review]

File: client/src/App.js [hunk 5]
```diff
@@ -1,24 +1,32 @@
...
-      </header>
-    </div>
-  );
+
+class App extends Component {
+
+    constructor(props) {
+  super(props);
+  this.state = { text: "" };
+}
```
Comment (ajwang0) [inline client/src/App.js]: Can you reformat this so it's aligned properly

File: client/src/App.test.js [hunk 6]
```diff
@@ -1,9 +1,27 @@
...
-  const linkElement = getByText(/learn react/i);
-  expect(linkElement).toBeInTheDocument();
+const renderComponent = () =>
+render(
+    <FetchMock
+    options={
+        {matcher: '/api/v1/test-servlet', method: 'GET', response: "Hello world"}
+    }
+    >
+    <App />
```
Comment (ajwang0) [inline client/src/App.test.js]: Can you indent <App /> so its more clear its a child of FetchMock

**Review by ajwang0 — DISMISSED** [review]

File: client/src/App.test.js [hunk 7]
```diff
@@ -1,9 +1,27 @@
 import React from 'react';
-import { render } from '@testing-library/react';
+import { render, screen, wait } from '@testing-library/react';
```
Comment (ajwang0) [inline client/src/App.test.js]: Also we can remove the screen dep

File: client/src/App.js (hunk 5 above)
Comment (essencec) [reply client/src/App.js]: Done

File: client/src/App.test.js (hunk 6 above)
Comment (essencec) [reply client/src/App.test.js]: Done

File: client/src/App.test.js (hunk 7 above)
Comment (essencec) [reply client/src/App.test.js]: Done

**Review by ajwang0 — APPROVED** [review]

**Pull request merged**